File: ios/telobike/telobike/TBPlacemarkAnnotation.m
//
//  TBPlacemarkAnnotation.m
//  telobike
//
//  Created by Elad Ben-Israel on 12/8/13.
//  Copyright (c) 2013 Elad Ben-Israel. All rights reserved.
//

#import "TBPlacemarkAnnotation.h"

@interface TBPlacemarkAnnotation ()

@property (strong, nonatomic) SVPlacemark* placemark;

@end

@implementation TBPlacemarkAnnotation

- (id)initWithPlacemark:(SVPlacemark *)placemark {
    self = [super init];
    if (self) {
        self.placemark = placemark;
    }
    return self;
}

- (CLLocationCoordinate2D)coordinate {
    return self.placemark.coordinate;
}

@end

File: ios/telobike/telobike/TBPlacemarkAnnotation.swift
//
//  TBPlacemarkAnnotation.swift
//  telobike
//
//  Created by Elad Ben-Israel on 11/17/14.
//  Copyright (c) 2014 Elad Ben-Israel. All rights reserved.
//

import UIKit

class TBPlacemarkAnnotation: NSObject, MKAnnotation {
    let placemark: SVPlacemark
    
    init(placemark: SVPlacemark) {
        self.placemark = placemark
    }

    var coordinate: CLLocationCoordinate2D {
        return self.placemark.coordinate
    }
    
    var title: String! {
        return self.placemark.formattedAddress
    }
}
